embedded-graphics/src/primitives/polyline.rs

```diff
@@ -0,0 +1,358 @@
+    type Item = Pixel<C>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        // Break if stroke width is zero
+        if self.style.stroke_width == 0 {
+            return None;
+        }
+
+        // Return none if stroke color is none
+        let stroke_color = self.style.stroke_color?;
+
+        self.line_iter
+            .next()
+            .map(|point| Pixel(point, stroke_color))
+    }
+}
+
+impl<'a, C: 'a> Drawable<C> for &Styled<Polyline<'a>, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    fn draw<D: DrawTarget<C>>(self, display: &mut D) -> Result<(), D::Error> {
+        display.draw_iter(self.into_iter())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{geometry::Size, pixelcolor::Rgb565, prelude::*};
+
+    // A "heartbeat" shaped polyline
+    const HEARTBEAT: [Point; 10] = [
+        Point::new(10, 64),
+        Point::new(50, 64),
+        Point::new(60, 44),
+        Point::new(70, 64),
+        Point::new(80, 64),
+        Point::new(90, 74),
+        Point::new(100, 10),
+        Point::new(110, 84),
+        Point::new(120, 64),
+        Point::new(300, 64),
+    ];
+
+    #[test]
+    fn positive_dimensions() {
+        let polyline = Polyline::new(&HEARTBEAT);
+
+        let bb = polyline.bounding_box();
+
+        assert_eq!(bb.top_left, Point::new(10, 10));
+        assert_eq!(bb.top_left + bb.size, Point::new(301, 85));
+        assert_eq!(bb.size, Size::new(291, 75));
+    }
+
+    #[test]
+    fn negative_dimensions() {
+        let mut negative: [Point; 10] = [Point::zero(); 10];
+
+        for (i, v) in HEARTBEAT.iter().enumerate() {
+            negative[i] = *v - Point::new(100, 100);
+        }
+
+        let polyline = Polyline::new(&negative);
+
+        let bb = polyline.bounding_box();
+
+        assert_eq!(bb.top_left, Point::new(-90, -90));
+        assert_eq!(bb.top_left + bb.size, Point::new(201, -15));
+        assert_eq!(bb.size, Size::new(291, 75));
+    }
+
+    #[test]
+    fn transformed_dimensions() {
+        let polyline = Polyline::new(&HEARTBEAT).translate(Point::new(-100, -100));
+
+        let bb = polyline.bounding_box();
+
+        assert_eq!(bb.top_left, Point::new(-90, -90));
+        assert_eq!(bb.top_left + bb.size, Point::new(201, -15));
+        assert_eq!(bb.size, Size::new(291, 75));
+    }
+
+    #[test]
+    #[should_panic]
+    fn one_point() {
+        let _polyline = Polyline::new(&[Point::zero()]);
+    }
+
+    // Ensure that polylines only draw 1px wide due to lack of support for line joiners. This test
+    // should fail when joiners are supported and should be removed then.
+    #[test]
+    fn one_px_wide_only() {
+        let polyline = Polyline::new(&HEARTBEAT);
+
+        let thick = polyline.into_styled(PrimitiveStyle::with_stroke(Rgb565::RED, 10));
+        let thin = polyline.into_styled(PrimitiveStyle::with_stroke(Rgb565::RED, 1));
+
+        assert!(thick.into_iter().eq(thin.into_iter()));
+    }
+}
```

Comment:
IMO some tests are missing:
* Drawing a polyline to `MockDisplay`
* Test that the `Points` iter doesn't return the points at the vertices twice